docs/proposals/namespace-deletion-protection/README.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Namespace deletion protection
+authors:
+- "@Vacant2333"
+reviewers:
+- "@XiShanYongYe-Chang"
+- "@zishen"
+- "@RainbowMango"
+approvers:
+- ""
+
+creation-date: 2023-07-17
+
+---
+
+# Namespace deletion protection
+
+## Summary
+
+In the Karmada control plane, the same Namespace manages the Namespaces of the same name in all member clusters. If a user mistakenly deletes the Namespace in the control plane, it means that the Namespaces in all member clusters will also be deleted, including various resources under it, such as Deployment, ConfigMap, Secret, etc. This impact is catastrophic.
+
+Therefore, we plan to provide a Namespace deletion protection mechanism. Users can prevent accidental deletion by adding a label to the target Namespace. When the user needs to delete it, they must remove the label to perform the deletion operation.
+
+## Motivation
+
+It is very necessary to avoid such mistakes. After implementing such a protection mechanism, we can extend this label to more types of resources, such as Deployment and Secret, etc., not only limited to protecting Namespace. In this way, it can be applicable in more scenarios.
+
+### Goals
+
+- Add or remove protection labels by karmadactl.
+- Prevent the user's deletion operation.
+
+## Proposal
+
+Protection Label Name: `karmada.io/deletion-protected`
+
+We can add a label `karmada.io/deletion-protected=true` to the namespace resource that needs protection. When a resource with this label is being deleted, we need to interrupt the user's operation request.
```

Comment:
Is this label added by the user or automatically added by the program?

Reply:
in my opnion, add the label is by user， i will update here